homeassistant/components/honeywell/config_flow.py

```diff
@@ -46,11 +58,14 @@ async def async_step_user(self, user_input=None):
 
     async def is_valid(self, **kwargs) -> bool:
         """Check if login credentials are valid."""
-        client = await self.hass.async_add_executor_job(
-            get_somecomfort_client, kwargs[CONF_USERNAME], kwargs[CONF_PASSWORD]
+        client = AIOSomecomfort.AIOSomeComfort(
+            kwargs[CONF_USERNAME],
+            kwargs[CONF_PASSWORD],
+            session=async_get_clientsession(self.hass),
         )
 
-        return client is not None
+        await client.login()
+        return True
```

Comment:
Since this method will always return True, the point of the return value is lost. Just remove the return statement and update the return value typing.